### PR TITLE
Add `FullHeightContent` component and fix `fullHeight` behavior of `MainContent`

### PR DESCRIPTION
**Pull request by jamesricky**
depends on #2754 

## Description

`FullHeightContent` is used to help components take advantage of all the available content height, e.g., when using a `DataGrid` inside `Tabs` already contained in a `MainContent` component.

Usage example for `FullHeightContent`:

```tsx
<MainContent>
    <RouterTabs>
        <RouterTab label="DataGrid Example" path="">
            <FullHeightContent>
                <DataGrid />
            </FullHeightContent>
        </RouterTab>
        <RouterTab label="Another tab" path="/another-tab">
            Content of another tab
        </RouterTab>
    </RouterTabs>
</MainContent>
```

Example where `MainContent` with `fullHeight` should be used, instead of `FullHeightContent`:

```tsx
<MainContent fullHeight>
    <DataGrid />
</MainContent>
```

## Example

-   [x] I have verified if my change requires an example

## Changeset

-   [x] I have verified if my change requires a changeset

## Related tasks and documents

https://vivid-planet.atlassian.net/browse/COM-1231

## Open TODOs/questions

-   [x] Merge parent PR https://github.com/vivid-planet/comet/pull/2754

### CODE REVIEW

File: packages/admin/admin/src/helpers/useTopOffset.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from "react";
+
+export const useTopOffset = (ref: React.RefObject<HTMLElement>) => {
```
Comment (johnnyomair) [inline packages/admin/admin/src/helpers/useTopOffset.ts:3]: Please add an import for RefObject here

File: storybook/src/admin/FullHeightContent.tsx
Comment (johnnyomair) [inline storybook/src/admin/FullHeightContent.tsx:1]: Please update this to the new CSF format

**Review by johnnyomair — COMMENTED** [review]
I'm not sure why we need this new component. Is it because we can't use MainContent inside a RouterTab?

And wouldn't a name like FillParent or FillContainer be more fitting?

**Comment by jamesricky** [comment]
> I'm not sure why we need this new component. Is it because we can't use MainContent inside a RouterTab?

IMO they are two separate use-cases:

- `MainContent` is used as the outer-most wrapper to contain a pages actual content and apply the content-spacing
- This new component can be use to give a child component access to the full remaining height, from the top of the current content, to the bottom of the browser window. This is usually a `DataGrid`, but could also be, for example, the block-preview/split-view. 

There often is an overlap, where both use-cases are needed, e.g., there is no content on a page, other than a `DataGrid`. 
In that case, content-spacing should be added and the child needs access to the full height. 
That's what `<MainContent fullHeight />` is useful for. 

```tsx
<Toolbar />
<MainContent fullHeight>
    <DataGrid />
</MainContent>
```

| Standalone DataGrid |
| --- |
| DataGrid standalone |


There are also many cases, where there is no overlap, e.g., when other content is shown above the `DataGrid`. 
This could be `Tabs` or anything else, like an `Alert` with some information/text. 

```tsx
<Toolbar />
<MainContent>
    <Tabs /> {/* Or Alert or any other content that should be indented by the content-spacing */}
    <FullHeightContent>
        <DataGrid />
    </FullHeightContent>
</MainContent>
```

| DataGrid with Tabs | DataGrid with Alert |
| --- | --- |
| DataGrid with Tabs | DataGrid with Alert |

The alternative would be to nest multiple instances of `MainContent` and disable the content-spacing. 
IMO this seems slightly confusing and not very intuitive, as the name "Main Content" and it's actual use-case don't really make sense anymore. 

```tsx
<Toolbar />
<MainContent>
    <Tabs /> {/* Or Alert or any other content that should be indented by the content-spacing */}
    <MainContent fullHeight disablePadding>
        <DataGrid />
    </MainContent>
</MainContent>
```

**Comment by jamesricky** [comment]
> And wouldn't a name like FillParent or FillContainer be more fitting?

I don't think so, as this component creates space for it's children, rather than filling an existing element 🤔

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**